src/Symfony/Bundle/FrameworkBundle/Command/SecretsUpdateKeyPairCommand.php

```diff
@@ -0,0 +1,116 @@
+    private $io;
+    private $secretsProcessor;
+
+    /**
+     * @param SecretVarProcessor $secretsProcessor
+     */
+    public function __construct(SecretVarProcessor $secretsProcessor)
+    {
+        $this->secretsProcessor = $secretsProcessor;
+        parent::__construct();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setDescription('Update or create your encrypted_secrets public/private key pair')
+            ->setHelp(<<<'HELP'
+The <info>%command.name%</info> updates your encrypted_secrets public/private key pair, or creates one if it does not already exist.
+If encrypted_secrets.enabled is set to true in your yaml config,
+
+  <info>php %command.full_name%</info>
+  
+will temporarily decrypt the values from the json secrets file set by encrypted_secrets.secrets_file using the public and private key pair specified by
+encrypted_secrets.public_key_file and encrypted_secrets.private_key_file. It will then generate a new key pair, re-encrypt your secrets, and update the public and private key files.
+If you haven't configured encrypted_secrets or wish to override these values, you can provide the required information via the public-key-file,
+private-key-file, and secrets-file options:
+
+  <info>php %command.full_name% --secrets-file ./config/secrets.jwe --public-key-file ./config/secrets.pub --private-key-file /etc/app-name/secret-key</info>
+
+If this is your first time using encrypted secrets, the specified files will be created for you. After running this command, use
+
+  <info>php bin/console secrets:add</info> or <info>php bin/console secrets:edit</info> to start adding your new secrets.
+  
+Always store your private key in a secure location outside of version control; you will not be able to recover your secrets without it.
+HELP
+            )
+            ->addOption('secrets-file', 's', InputOption::VALUE_REQUIRED, 'The file to edit encrypted secrets from')
+            ->addOption('public-key-file', 'p', InputOption::VALUE_REQUIRED, 'The 32 byte public key file to be used for encryption.')
+            ->addOption('private-key-file', 'x', InputOption::VALUE_REQUIRED, 'The 32 byte private key file to be used for decryption.')
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->io = new SymfonyStyle($input, $output);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $secretsLocation = $input->getOption('secrets-file');
+        $pubKeyLocation = $input->getOption('public-key-file');
+        $privateKeyLocation = $input->getOption('private-key-file');
+
+        if (isset($secretsLocation) && !file_exists($secretsLocation)) {
+            JweHandler::initSecretsFile($secretsLocation);
+        }
+
+        if ($this->secretsProcessor->isSecretsLookupEnabled()) {
```

Comment:
Hmm. I think there's a bit too much flexiblity here. It looks like I can: (A) configure my secrets setup in config OR (B) not configure it, but pass all the options here. If there is only one way to decrypt and use the secrets at runtime, why make there multiple ways to encrypt secrets here?

Reply:
Yeah I agree, a lot of this is an artifact of the initial implementation and restrictions that I thought people would have managing secrets per environment. If configuring the commands through the yaml files is acceptable to devs (and I don't see why it wouldn't be now) it would clean up a ton of this logic